stablehlo/reference/Ops.cpp

```diff
@@ -550,62 +550,68 @@ Tensor evalDotGeneralOp(const Tensor &lhs, const Tensor &rhs,
                         const Axes &rhsContractingDimensions,
                         ShapedType resultType) {
   Tensor result(resultType);
-  Axes lhsNonBatchingNonContractingDims;
+  Axes lhsResultDims;
   for (auto i = 0; i < lhs.getType().getRank(); ++i)
     if (!llvm::is_contained(lhsBatchingDimensions, i) &&
         !llvm::is_contained(lhsContractingDimensions, i))
-      lhsNonBatchingNonContractingDims.push_back(i);
+      lhsResultDims.push_back(i);
 
-  Axes rhsNonBatchingNonContractingDims;
+  Axes rhsResultDims;
   for (auto i = 0; i < rhs.getType().getRank(); ++i)
     if (!llvm::is_contained(rhsBatchingDimensions, i) &&
         !llvm::is_contained(rhsContractingDimensions, i))
-      rhsNonBatchingNonContractingDims.push_back(i);
-
-  Sizes contractingDimSizes;
-  int64_t totalContractingSize = 1;
-  for (uint64_t i = 0; i < lhsContractingDimensions.size(); ++i) {
-    contractingDimSizes.push_back(
-        lhs.getType().getShape()[lhsContractingDimensions[i]]);
-    totalContractingSize *=
-        lhs.getType().getShape()[lhsContractingDimensions[i]];
-  }
```

Comment:
I don't think we need these variables: `contractingDimSizes` can be replaced by `getShape()[lhsContractingDimensions[i]]`, and `totalContractingSize` can be replaced by `break`.